xpdacq/calib.py

```diff
@@ -122,35 +122,103 @@ def run_calibration(exposure=60, dark_sub=True, calibrant_file=None,
+
     Parameters
     ----------
-    img : ndarray
-        image that going to be masked
-    geometry_object : pyFAI.geometry.Geometry
-        The pyFAI description of the detector orientation or any
-        subclass of pyFAI.geometry.Geometry class
-    save_name : str, optional
-        full path for this mask going to be saved. if it is None, 
-        only mask will be returned, no file will be saved locally.
+    exposure : float, optional
+        exposure time of this scan. default is 60s.
+    dark_sub : bool, optional
+        turn on/off of dark subtraction. default is True.
+    sample_name : str, optional
+        name of sample that new mask is going to be generated from.
+        default is 'mask_target'
+    calib_dict : dict, optional
+        dictionary with parameters for geometry correction
+        software. default is read out from glbl attribute (parameters
+        from the most recent calibration)
     mask_dict : dict, optional
-        dictionary for arguments in masking function. for more details, 
-        please see docstring of xpdan.tools.mask_img
+        dictionary for arguments in masking function. for more details,
+        please check docstring from ``xpdan.tools.mask_img``
+    save_name : str, optional
+        full path for this mask going to be saved. if it is None, only
+        the mask object will be returned, no file will be saved locally.
+
+    Note
+    ----
+    current software dealing with geometry correction is ``pyFAI``
 
     See also
     --------
     xpdan.tools.mask_img
     """
-    mask = mask_img(img, geometry_object, **mask_dict)
+
+    _check_obj(_REQUIRED_OBJ_LIST)
+    ips = get_ipython()
+    bto = ips.ns_table['user_global']['bt']
+    prun = ips.ns_table['user_global']['prun']
+
+    # default behavior
+    if sample_name is None:
+        sample_name = 'mask_target'
+
+    if mask_dict is None:
+        mask_dict = glbl.mask_dict
+
+    if calib_dict is None:
+        calib_dict = glbl.calib_config_dict
+
+    # setting up geometry parameters
+    ai = AzimuthalIntegrator()
+    ai.setPyFAI(**calib_dict)
+
+    # scan
+    mask_collection_uid = str(uuid.uuid4())
+    mask_builder_dict = {'sample_name':sample_name,
+                        'sample_composition':{sample_name :1},
+                        'is_mask': True,
+                        'mask_collection_uid': mask_collection_uid}
+    sample = Sample(bto, mask_builder_dict)
+    prun_uid = prun(sample, ScanPlan(bto, ct, exposure))
+    light_header = glbl.db[-1]
+    if dark_sub:
+        dark_uid = light_header.start['sc_dk_field_uid']
+        dark_header = glbl.db[dark_uid]
+        dark_img = np.asarray(glbl.db.get_images(dark_header,
+                                glbl.det_image_field)).squeeze()
+    for ev in glbl.db.get_events(light_header, fill=True):
+        img = ev['data'][glbl.det_image_field]
+        if dark_sub:
+            img -= dark_img
+
+    print("INFO: masking you image ....")
```

Comment:
Please delete this message.  It is kind of obvious that a mask is being created when you run ``mask_builder``